app/services/fact_parser.rb

```diff
@@ -33,18 +33,37 @@ def operatingsystem
   end
 
   def environment
```

Comment:
this very soon becomes the puppet plugin only and shouldn't be in the abstract parser, puppet environment will no longer be a core resource

before that happens, it can potentially live in the puppet_fact_parser, but again very shortly the codewill need to live in the foreman_puppet plugin, should we have some formal parsers extensions points to avoid including modules or monkeypatching?